### PR TITLE
added collector for approval event erc20

### LINKED ISSUE #10: Read `Approval` events from ERC-20 contracts.

### PR DESCRIPTION
**Pull request by evictorero**
- Added new collector for Approvals Events
 - Refactored extracting common methods and structs for all ERC20 Events
 - Added some steps that I am using to Makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,3 +15,12 @@ mixin: lint-mixin
 
 lint-mixin:
 	mixtool lint mixin/mixin.libsonnet
+
+build:
+	docker-compose build
+
+run:
```
Comment (thepalbi) [inline Makefile]: nit: 
```suggestion
start:
```

File: internal/collectors/contracts/erc20/approval_event.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+			return nil, errors.Wrap(err, "failed to create ERC20 approval evt collector")
+		}
+		info, err := getContractInfo(contractAddress, client)
+		if err != nil {
+			return nil, err
+		}
+
+		log.Printf("Got info for %s, symbol %s\n", info.Address, info.Symbol)
+		clients[info] = filterer
+	}
```
Comment (thepalbi) [inline internal/collectors/contracts/erc20/approval_event.go]: This bit could be extracted as well right?

File: internal/collectors/contracts/erc20/approval_event.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+			defer wg.Done()
+			col.doCollect(ch, currentBlockNumber, contrInfo, client)
+		}(contrInfo, client)
+	}
+
+	wg.Wait()
+
+	// Improve error model, this will advance last seen block even if some client fails
+	col.lastQueriedBlock = currentBlockNumber
+}
```
Comment (thepalbi) [inline internal/collectors/contracts/erc20/approval_event.go:109]: Also this part can be extracted somehow, so each implementation only implements `doCollect`?

**Comment by thepalbi** [comment]
@evictorero added some comments regarding some other parts we could extract / generalize. I don't have a strong opinion about them. Feel free to merge this PR and in a follow up extract them, or leave a comment with a TODO

**Review by thepalbi — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (evictorero) [reply Makefile]: Done

File: internal/collectors/contracts/erc20/approval_event.go (hunk 2 above)
Comment (evictorero) [reply internal/collectors/contracts/erc20/approval_event.go]: Yes, easily. Done.

File: internal/collectors/contracts/erc20/approval_event.go (hunk 3 above)
Comment (evictorero) [reply internal/collectors/contracts/erc20/approval_event.go:109]: This one would require more changes, we can do it in the future

**Pull request merged**